packages/utils/src/ts-eslint/Rule.ts

```diff
@@ -632,33 +632,48 @@ interface RuleModule<
    */
   create(context: Readonly<RuleContext<TMessageIds, TOptions>>): TRuleListener;
 }
-type AnyRuleModule = RuleModule<string, readonly unknown[]>;
+export type AnyRuleModule = RuleModule<string, readonly unknown[]>;
 
-type RuleCreateFunction<
+/**
+ * A loose definition of the RuleModule type for use with configs. This type
+ * intended to relax validation of types so that we can have basic validation
+ * without being overly strict about nitty gritty details matching.
+ *
+ * For example the plugin might be declared using an old version of our types or
+ * they might use the DefinitelyTyped eslint types. Ultimately we don't need
+ * super strict validation in a config - a loose shape match is "good enough" to
+ * help validate the config is correct.
+ *
+ * See also LooseParserModule, LooseProcessorModule
+ */
+export type LooseRuleDefinition =
+  // TODO - ESLint v9 will remove support for RuleCreateFunction
+  | LooseRuleCreateFunction
+  | {
+      meta?: object;
+      create: LooseRuleCreateFunction;
+    };
+/*
+eslint-disable-next-line @typescript-eslint/no-explicit-any --
+intentionally using `any` to allow bi-directional assignment (unknown and
+never only allow unidirectional)
+*/
+export type LooseRuleCreateFunction = (context: any) => Record<
+  string,
+  /*
+  eslint-disable-next-line @typescript-eslint/ban-types --
+  intentionally use Function here to give us the basic "is a function" validation
+  without enforcing specific argument types so that different AST types can still
+  be passed to configs
+  */
+  Function | undefined
+>;
+
+export type RuleCreateFunction<
   TMessageIds extends string = never,
   TOptions extends readonly unknown[] = unknown[],
 > = (context: Readonly<RuleContext<TMessageIds, TOptions>>) => RuleListener;
-type AnyRuleCreateFunction = RuleCreateFunction<string, readonly unknown[]>;
-
-export {
-  AnyRuleCreateFunction,
-  AnyRuleModule,
-  CodePath,
-  CodePathFunction,
-  CodePathSegment,
-  ReportDescriptor,
-  ReportDescriptorMessageData,
-  ReportFixFunction,
-  ReportSuggestionArray,
-  RuleContext,
-  RuleCreateFunction,
-  RuleFix,
-  RuleFixer,
-  RuleFunction,
-  RuleListener,
-  RuleListenerExtension,
-  RuleMetaData,
-  RuleMetaDataDocs,
-  RuleModule,
-  SharedConfigurationSettings,
-};
```

Comment:
[Non-Actionable] That reminds me, I totally forgot about https://github.com/typescript-eslint/typescript-eslint/issues/5871 ...